pkg/plugins/volume/ceph/size.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package ceph
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/ceph/go-ceph/rbd"
+	"github.com/go-logr/logr"
+	"github.com/ironcore-dev/libvirt-provider/pkg/api"
+)
+
+func createKeyFile(imageName string, key string) (string, func() error, error) {
+	noOpCleanup := func() error { return nil }
+	file, err := os.CreateTemp("", fmt.Sprintf("%s-*", imageName))
+	if err != nil {
+		return "", noOpCleanup, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	cleanup := func() error {
```

Comment:
Why not returning the file handler here and do the `defer os.RemoveFile(tempFile)` whenever it is needed.

Reply:
do we need file handler here? it can do more problems with multiple goroutines we can easily hit file descriptor limits and rbd command require path to file not file descriptor/handler.

Reply:
or what do you mean file handler? File path or File structure?

Reply:
Actually, I would like to keep it like this since currently the function is self contained. Moving `defer os.RemoveFile(tempFile)` to the outside makes the caller to know that some cleanup is needed and in case of using the function at multiple places spreads this logic around the project.

Reply:
i agree with @lukasfrank, current solution looks fine for me.